docs/DeveloperGuide.md

```diff
@@ -1,13 +1,413 @@
+| date            | LocalDate     | The date of the expense              |
+
+The BudgetBuddy class then calls the `execute()` method of the `AddExpenseCommand` object which uses the following
+method in the `ExpenseManager` class to add the expense:
+
+| Method              | Return Type | Relevance                                 |
+|---------------------|-------------|-------------------------------------------|
+| addExpense(expense) | void        | Add new expense to the list of `expenses` |                                 
+
+A `RemainingBudgetManager` object will be created to find the budget remaining for the given month and category. 
+Finally, the acknowledgement message along with the budget remaining is displayed to the user using the `Ui` class 
+`displayToUser()` method.
+
+The following UML Sequence diagram shows how the Parser works to obtain the relevant inputs for the Add Expense Feature:
+The scenario of invalid input and sequence within `RemainingBudgetManager` are omitted to reduce complexity.
+![AddExpenseSequenceDiagram.drawio.png](diagrams/AddExpenseSequenceDiagram.drawio.png)
+
+
+#### 4.2 Add Budget Feature
+The Add Budget feature enables users to add budgets for different categories. This functionality is controlled by the 
+AddBudgetCommand class, which is produced by the Parser class based on user input. The AddBudgetCommand class uses an 
+AddBudgetValidator object to validate the provided amount, category, and date, and then performs the budget addition 
+if valid. Below is the relevance of these attributes:
+
+| Class Attribute | Variable Type | Relevance                                   |
+|-----------------|---------------|---------------------------------------------|
+| amount          | double        | The budget amount to be added               |
+| category        | Category      | The category of budget to be added          |
+| date            | YearMonth     | The specific month and year of budget added |
+
+The BudgetBuddy class then calls the `execute()` method of the `AddBudgetCommand` object which uses the following 
+method in the `BudgetManager` class to add a new budget created by the `Budget` class to the `budgets` list 
+if it does not exist:
+
+| Method            | Return Type | Relevance                               |
+|-------------------|-------------|-----------------------------------------|
+| addBudget(budget) | void        | Add new budget to the list of `budgets` |                                 
+
+Then, `addAmount(category, amount)` method in the `Budget` class is also called to add the amount to the budget. 
+Finally, the result is displayed to the user using the `Ui` class `displayToUser()` method.
+
+The following UML Sequence diagram shows how the Parser works to obtain the relevant inputs 
+for the Add Budget Feature:
+![AddBudgetSequenceDiagram.drawio.png](diagrams/AddBudgetSequenceDiagram.drawio.png)
```

Comment:
Is writing return on the dashed lines necessary? Should it be blank instead?
![image](https://github.com/user-attachments/assets/8d12c32b-6205-4108-bcf9-d21ddfdc8141)